File: algorithms/1_easy/F/00997_find_the_town_judge/solution.cpp
class Solution {
public:
    static int findJudge(int n, vector<vector<int>>& trust) {
        const size_t len{static_cast<size_t>(n + 1)};
        vector<int> count(len, 0);

        for (const auto& t : trust) {
            count[static_cast<size_t>(t[0])] -= 1;
            count[static_cast<size_t>(t[1])] += 1;
        }

        for (size_t i{1}; i < len; ++i) {
            if (count[i] == n - 1) {
                return static_cast<int>(i);
            }
        }

        return -1;
    }
};

File: algorithms/1_easy/F/00997_find_the_town_judge/solution.rs
impl Solution {
    pub fn find_judge(n: i32, trust: Vec<Vec<i32>>) -> i32 {
        let len = n as usize + 1;
        let mut count = vec![0_i32; len];

        for t in trust.iter() {
            count[t[0] as usize] -= 1;
            count[t[1] as usize] += 1;
        }

        for i in 1..len {
            if count[i] == n - 1 {
                return i as i32;
            }
        }

        -1
    }
}
